magento/magento2
# Improve product edit page loading speed

### Description (*)

Page loading speed for products with many custom options or bundle products with multiple options with many selections is slow.
The more custom options a product has and the more options and selections a bundle product has, the slower the speed becomes.

In the issue addressed below, several experiences were shared about how slow this can get.

With the changes included in this Pull Request, I was able to reduce the page load time for one of my bundle products (10 options, approx. 160 selections in total) from about 1min 20sec to 35sec.

### Related Pull Requests
https://github.com/magento-gl/magento2-page-builder/pull/53


### Fixed Issues (if relevant)

1. Fixes magento/magento2#29409

### Manual testing scenarios (*)

1. Create a bundle product with one option and add one selection
2. Load the edit page and measure the time until the page becomes fully responsive (JavaScript finishes)
3. Add more options (about 5) and several selections (about 20) to each option
4. Load the page and measure the time again

Another scenario would be to add more and more custom options to a product and note that it takes longer to load the edit page.

### Questions or comments


### Contribution checklist (*)
